### PR TITLE
Final bits of GameEngine::Init

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

**Review by tomsons26 — APPROVED** [review]

File: src/game/common/system/kindof.cpp [hunk 1]
```diff
@@ -139,3 +139,25 @@ const char *BitFlags<KINDOF_COUNT>::s_bitNamesList[]{ "OBSTACLE",
...
+    KINDOFMASK_FS.Set(KINDOF_FS_BLACK_MARKET, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_SUPPLY_CENTER, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_STRATEGY_CENTER, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_FAKE, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_INTERNET_CENTER, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_ADVANCED_TECH, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_BARRACKS, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_WARFACTORY, true);
+    KINDOFMASK_FS.Set(KINDOF_FS_AIRFIELD, true);
+}
```
Comment (xezon) [inline src/game/common/system/kindof.cpp]: eol

File: src/game/common/system/savegame/gamestate.cpp [hunk 2]
```diff
@@ -22,6 +22,50 @@
 GameState *g_theGameState = nullptr;
 #endif
 
+SaveGameInfo::SaveGameInfo() : m_missionNumber(0), m_saveFileType(SAVE_TYPE_UNK) {}
+
+GameState::GameState() : m_availableGames(nullptr), m_isLoading(false) {}
+
+GameState::~GameState()
+{
+    for (int i = 0; i < 3; i++) {
```
Comment (xezon) [inline src/game/common/system/savegame/gamestate.cpp]: i < ARRAY_SIZE(m_snapShotBlocks)

File: src/game/logic/system/damage.h [hunk 3]
```diff
@@ -80,3 +80,13 @@ class DamageInfo : public SnapShot
...
+
+#ifdef GAME_DLL
+extern BitFlags<DAMAGE_NUM_TYPES> &DAMAGE_TYPE_FLAGS_ALL;
+#else
+extern BitFlags<DAMAGE_NUM_TYPES> DAMAGE_TYPE_FLAGS_ALL;
+#endif
+
+extern BitFlags<DAMAGE_NUM_TYPES> DAMAGE_TYPE_FLAGS_NONE;
+
+void Init_Damage_Type_Masks();
```
Comment (xezon) [inline src/game/logic/system/damage.h]: eol

File: src/game/common/system/kindof.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/common/system/kindof.cpp]: fixed.

File: src/game/common/system/savegame/gamestate.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/common/system/savegame/gamestate.cpp]: fixed.

File: src/game/logic/system/damage.h (hunk 3 above)
Comment (jonwil) [reply src/game/logic/system/damage.h]: fixed.

**Pull request merged**